sdk/iot/core/tests/cmocka/az_iot_hub_client_tests.c

```diff
@@ -21,3 +36,175 @@ void test_az_iot_hub_client_get_default_options_succeed(void** state)
   assert_true(az_span_is_content_equal(options.module_id, AZ_SPAN_NULL));
   assert_true(az_span_is_content_equal(options.user_agent, AZ_SPAN_NULL));
 }
+
+void test_az_iot_hub_client_init_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  assert_string_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id));
+  assert_string_equal(TEST_DEVICE_HOSTNAME_STR, az_span_ptr(client._internal.iot_hub_hostname));
+}
+
+void test_az_iot_hub_client_init_custom_options_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options;
+  options.module_id = AZ_SPAN_FROM_STR(TEST_MODULE_ID);
+  options.user_agent = AZ_SPAN_FROM_STR(TEST_USER_AGENT);
+  assert_true(
```

Comment:
>assert_true [](start = 2, length = 11)

Assert_int_equal, helps to show what was the return value of function, when assert failed.

Reply:
good call. I'll make a PR updating the other code too.